kb-survey/src/app/survey/survey.component.ts

```diff
@@ -101,30 +135,126 @@ export class SurveyComponent implements OnInit {
   }
 
   fetchSurveyDetails() {
+    this.showSpinner = true;
     this.baseApiService
       .post(
         urlConfig.surveyDetailsURL + `?solutionId=${this.solutionId}`,
         this.profileDetails
       )
       .subscribe((res: any) => {
-        this.assessmentResult = res.result;
+        if (res?.message == "Survey details fetched successfully") {
+          this.assessmentResult = res.result;
+        }
+        else if (res?.message == "Could not found solution details") {
+          this.responseService.setResponse("Could not found solution details.");
+          this.router.navigateByUrl('/response');
+        } else {
+          const confirmationParams = {
+            title: "Error",
+            message: "Something went wrong, try again",
+            timer: 3000,
+            actionBtns: false,
+            btnLeftLabel: "",
+            btnRightLabel: ""
+          };
+
+          this.openConfirmationDialog(confirmationParams)
+        }
+        this.showSpinner = false;
         console.log(this.assessmentResult);
       });
   }
 
-  submitOrSaveEvent(event: any): void {
+  async submitOrSaveEvent(event: any) {
     const evidenceData = { ...event.detail.data, status: event.detail.status };
-    this.baseApiService
-      .post(
-        urlConfig.surveySubmissionURL +
+    if (event?.detail?.status == "submit") {
+      const confirmationParams = {
+        title: "Confirmation",
+        message: "Are you sure you want to submit survey?",
+        timer: 'fasle',
+        actionBtns: true,
+        btnLeftLabel: "Cancel",
+        btnRightLabel: "Confirm"
+      };
+      const response = await this.openConfirmationDialog(confirmationParams)
+      if (response) {
+        this.showSpinner = true;
+        this.baseApiService
+          .post(
+            urlConfig.surveySubmissionURL +
+            this.assessmentResult.assessment.submissionId,
+            {
+              ...this.profileDetails,
+              evidence: evidenceData,
+            }
+          )
+          .subscribe((res: any) => {
+            window.postMessage(res, '*');
+            this.responseService.setResponse("Thank you, your survey has been Submited.");
+            this.showSpinner = false;
+            this.router.navigateByUrl('/response');
+          },
+            (err: any) => {
+              const confirmationParams = {
+                title: "Error",
+                message: "Something went wrong, try again",
+                timer: 3000,
+                actionBtns: false,
+                btnLeftLabel: "",
+                btnRightLabel: ""
+              };
+
+              this.openConfirmationDialog(confirmationParams)
+            }
+          );
+      }
+    } else {
+      this.showSpinner = true;
+
+      this.baseApiService
```

Comment:
This is not resolved yet, you're rewriting entire logic for save and submit instead find a way to use the same logic for both cases